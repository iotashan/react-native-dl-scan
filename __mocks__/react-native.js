/* eslint-disable @typescript-eslint/no-unused-vars */

const React = require('react');
const RN = jest.requireActual('react-native');

// Mock all the needed modules
const mockAnimatedValue = () => ({
  setValue: jest.fn(),
  setOffset: jest.fn(),
  flattenOffset: jest.fn(),
  extractOffset: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  removeAllListeners: jest.fn(),
  stopAnimation: jest.fn(),
  resetAnimation: jest.fn(),
  interpolate: jest.fn(),
  animate: jest.fn(),
  _startListeningToNativeValueUpdates: jest.fn(),
  _stopListeningForNativeValueUpdates: jest.fn(),
});

const mockAnimatedTimingConfig = () => ({
  start: jest.fn((callback) => callback && callback({ finished: true })),
  stop: jest.fn(),
  reset: jest.fn(),
});

const Platform = {
  OS: 'ios',
  select: jest.fn((obj) => obj.ios || obj.default),
  Version: '14.0',
  constants: {
    isTesting: true,
    osVersion: '14.0',
    reactNativeVersion: {
      major: 0,
      minor: 79,
      patch: 0,
    },
    systemName: 'iOS',
  },
  isDisableAnimations: false,
};

const Animated = {
  Value: jest.fn(mockAnimatedValue),
  View: 'Animated.View',
  Text: 'Animated.Text',
  Image: 'Animated.Image',
  ScrollView: 'Animated.ScrollView',
  createAnimatedComponent: jest.fn((component) => component),
  timing: jest.fn(mockAnimatedTimingConfig),
  spring: jest.fn(mockAnimatedTimingConfig),
  sequence: jest.fn(mockAnimatedTimingConfig),
  parallel: jest.fn(mockAnimatedTimingConfig),
  decay: jest.fn(mockAnimatedTimingConfig),
  delay: jest.fn(mockAnimatedTimingConfig),
  loop: jest.fn(mockAnimatedTimingConfig),
  event: jest.fn(),
  add: jest.fn(),
  subtract: jest.fn(),
  divide: jest.fn(),
  multiply: jest.fn(),
  modulo: jest.fn(),
  diffClamp: jest.fn(),
};

// Create proper component mocks
const TouchableOpacity = ({ children, ...props }) => 
  React.createElement('TouchableOpacity', props, children);

const TouchableHighlight = ({ children, ...props }) => 
  React.createElement('TouchableHighlight', props, children);

const TouchableWithoutFeedback = ({ children, ...props }) => 
  React.createElement('TouchableWithoutFeedback', props, children);

module.exports = {
  ...RN,
  Platform,
  Animated,
  // Mock Touchable components
  TouchableOpacity,
  TouchableHighlight,
  TouchableWithoutFeedback,
  // Mock other components
  View: ({ children, ...props }) => React.createElement('View', props, children),
  Text: ({ children, ...props }) => React.createElement('Text', props, children),
  Image: (props) => React.createElement('Image', props),
  ScrollView: ({ children, ...props }) => React.createElement('ScrollView', props, children),
  FlatList: ({ children, ...props }) => React.createElement('FlatList', props, children),
  // Mock utilities
  StyleSheet: {
    create: (styles) => styles,
    flatten: jest.fn(),
    compose: jest.fn(),
  },
  Dimensions: {
    get: jest.fn(() => ({ width: 375, height: 667 })),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  },
  PixelRatio: {
    get: jest.fn(() => 2),
    getFontScale: jest.fn(() => 1),
    getPixelSizeForLayoutSize: jest.fn((size) => size * 2),
    roundToNearestPixel: jest.fn((size) => size),
  },
  // Mock APIs
  Vibration: {
    vibrate: jest.fn(),
  },
  AccessibilityInfo: {
    announceForAccessibility: jest.fn(),
    isReduceMotionEnabled: jest.fn(() => Promise.resolve(false)),
    isScreenReaderEnabled: jest.fn(() => Promise.resolve(false)),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  },
  // Mock NativeModules
  NativeModules: {
    ...RN.NativeModules,
  },
};