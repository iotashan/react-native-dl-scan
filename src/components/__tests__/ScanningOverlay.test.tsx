import { render } from '@testing-library/react-native';
import { ScanningOverlay } from '../ScanningOverlay';

// Mock react-native-reanimated
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};
  return Reanimated;
});

const defaultProps = {
  mode: 'auto' as const,
  isScanning: false,
  detectionState: 'idle' as const,
  orientation: 'portrait' as const,
};

describe('ScanningOverlay', () => {
  it('renders correctly in auto mode', () => {
    const { toJSON } = render(<ScanningOverlay {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly in barcode mode', () => {
    const { getByText } = render(
      <ScanningOverlay {...defaultProps} mode="barcode" />
    );
    expect(getByText(/Position the barcode/)).toBeTruthy();
  });

  it('renders correctly in OCR mode', () => {
    const { getByText } = render(
      <ScanningOverlay {...defaultProps} mode="ocr" />
    );
    expect(getByText(/Position the front of your license/)).toBeTruthy();
  });

  it('displays custom instruction text', () => {
    const instructionText = 'Custom instruction';
    const { getByText } = render(
      <ScanningOverlay {...defaultProps} instructionText={instructionText} />
    );
    expect(getByText(instructionText)).toBeTruthy();
  });

  it('renders in landscape orientation', () => {
    const { toJSON } = render(
      <ScanningOverlay {...defaultProps} orientation="landscape" />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('hides guides when showGuides is false', () => {
    const { queryByTestId } = render(
      <ScanningOverlay {...defaultProps} showGuides={false} />
    );
    // Corners should not be rendered
    expect(queryByTestId('corner-top-left')).toBeNull();
  });

  it('renders success state correctly', () => {
    const { toJSON } = render(
      <ScanningOverlay
        {...defaultProps}
        detectionState="success"
        animateSuccess={true}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders error state correctly', () => {
    const { toJSON } = render(
      <ScanningOverlay
        {...defaultProps}
        detectionState="error"
        animateError={true}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onOverlayPress when pressed', () => {
    const onOverlayPress = jest.fn();
    render(
      <ScanningOverlay {...defaultProps} onOverlayPress={onOverlayPress} />
    );

    // Would need to add testID to the touchable area in the component
    // For now, this is a placeholder test
    expect(onOverlayPress).not.toHaveBeenCalled();
  });
});

describe('ScanningOverlay animations', () => {
  it('starts pulse animation when scanning', () => {
    const { rerender } = render(
      <ScanningOverlay {...defaultProps} isScanning={false} />
    );

    rerender(<ScanningOverlay {...defaultProps} isScanning={true} />);

    // Animation values would be tested here if we had access to the shared values
    // This is more of an integration test
  });

  it('shows sweep line in barcode mode when scanning', () => {
    const { toJSON } = render(
      <ScanningOverlay {...defaultProps} mode="barcode" isScanning={true} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
