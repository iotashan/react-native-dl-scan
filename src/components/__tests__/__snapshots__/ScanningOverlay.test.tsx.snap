// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanningOverlay animations shows sweep line in barcode mode when scanning 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "height": 401.448362720403,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
            },
            {
              "height": 120.4345088161209,
              "marginBottom": 281.0138539042821,
              "width": 637.5,
            },
          ]
        }
      />
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        {
          "height": 120.4345088161209,
          "marginBottom": 281.0138539042821,
          "width": 637.5,
        },
        {
          "transform": [
            {
              "scale": 1,
            },
            {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 8,
            "left": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderTopRightRadius": 8,
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "left": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomRightRadius": 8,
            "borderLeftWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "right": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 12,
            "borderWidth": 2,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "opacity": 0.5,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "height": 2,
            "opacity": 0.8,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 5,
            "width": "90%",
          },
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "translateY": undefined,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      {
        "paddingHorizontal": 40,
        "position": "absolute",
        "top": 125,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFD700",
          "fontSize": 14,
          "fontWeight": "500",
          "textAlign": "center",
          "textShadowColor": "rgba(0, 0, 0, 0.75)",
          "textShadowOffset": {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 3,
        }
      }
    >
      Position the barcode on the back of your license
    </Text>
  </View>
</View>
`;

exports[`ScanningOverlay renders correctly in auto mode 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "height": 401.448362720403,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
            },
            {
              "height": 401.448362720403,
              "width": 637.5,
            },
          ]
        }
      />
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        {
          "height": 401.448362720403,
          "width": 637.5,
        },
        {
          "transform": [
            {
              "scale": 1,
            },
            {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 8,
            "left": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderTopRightRadius": 8,
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "left": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomRightRadius": 8,
            "borderLeftWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "right": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ScanningOverlay renders error state correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "height": 401.448362720403,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
            },
            {
              "height": 401.448362720403,
              "width": 637.5,
            },
          ]
        }
      />
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        {
          "height": 401.448362720403,
          "width": 637.5,
        },
        {
          "transform": [
            {
              "scale": 1,
            },
            {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 8,
            "left": 0,
            "top": 0,
          },
          {
            "borderColor": "#F44336",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderTopRightRadius": 8,
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#F44336",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "left": 0,
          },
          {
            "borderColor": "#F44336",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomRightRadius": 8,
            "borderLeftWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "right": 0,
          },
          {
            "borderColor": "#F44336",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ScanningOverlay renders in landscape orientation 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "height": 401.448362720403,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
            },
            {
              "height": 637.5,
              "width": 401.448362720403,
            },
          ]
        }
      />
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        {
          "height": 637.5,
          "width": 401.448362720403,
        },
        {
          "transform": [
            {
              "scale": 1,
            },
            {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 8,
            "left": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderTopRightRadius": 8,
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "left": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomRightRadius": 8,
            "borderLeftWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "right": 0,
          },
          {
            "borderColor": "#FFFFFF",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ScanningOverlay renders success state correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "height": 401.448362720403,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "transparent",
            },
            {
              "height": 401.448362720403,
              "width": 637.5,
            },
          ]
        }
      />
      <View
        style={
          {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    pointerEvents="none"
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "position": "absolute",
        },
        {
          "height": 401.448362720403,
          "width": 637.5,
        },
        {
          "transform": [
            {
              "scale": 1,
            },
            {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": 8,
            "left": 0,
            "top": 0,
          },
          {
            "borderColor": "#4CAF50",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderTopRightRadius": 8,
            "right": 0,
            "top": 0,
          },
          {
            "borderColor": "#4CAF50",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "left": 0,
          },
          {
            "borderColor": "#4CAF50",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderWidth": 3,
            "height": 30,
            "position": "absolute",
            "width": 30,
          },
          {
            "borderBottomRightRadius": 8,
            "borderLeftWidth": 0,
            "borderTopWidth": 0,
            "bottom": 0,
            "right": 0,
          },
          {
            "borderColor": "#4CAF50",
          },
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "opacity": 0,
            "transform": [
              {
                "scale": 0,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#4CAF50",
            "borderRadius": 40,
            "height": 80,
            "justifyContent": "center",
            "width": 80,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 40,
              "fontWeight": "bold",
            }
          }
        >
          ✓
        </Text>
      </View>
    </View>
  </View>
</View>
`;
